pkg/agent/manager/cache/cache.go

```diff
@@ -174,7 +190,9 @@ func (c *Cache) SubscribeToWorkloadUpdates(selectors []*common.Selector) Subscri
 	return sub
 }
 
-func (c *Cache) Update(update *Update, checkSVID func(*common.RegistrationEntry, *common.RegistrationEntry, *X509SVID)) {
+// UpdateEntries updates provided registration entries and bundles in cache, based in updates all subscribers are notified about updates.
+// In case checkSVID is provided, it is used to verify svid rotation, returning `true` in case it is marked as `stale`.
+func (c *Cache) UpdateEntries(update *UpdateEntries, checkSVID func(*common.RegistrationEntry, *common.RegistrationEntry, *X509SVID) bool) {
```

Comment:
```suggestion
// UpdateEntries updates the cache with the provided registration entries and bundles and
// notifies impacted subscribers. The checkSVID callback, if provided, is used to determine
// if the SVID for the entry is stale, or otherwise in need of rotation. Entries marked stale
// through the checkSVID callback are returned from GetStaleEntries() until the SVID is
// updated through a call to UpdateSVIDs.
func (c *Cache) UpdateEntries(update *UpdateEntries, checkSVID func(*common.RegistrationEntry, *common.RegistrationEntry, *X509SVID) bool) {
```